src/main/java/com/nvs/store/service/CartService.java

```diff
@@ -0,0 +1,202 @@
+package com.nvs.store.service;
+
+import com.nvs.store.dto.*;
+import com.nvs.store.models.cart.Cart;
+import com.nvs.store.models.cart.CartItem;
+import com.nvs.store.models.product.Product;
+import com.nvs.store.models.user.User;
+import com.nvs.store.repository.*;
+import com.nvs.store.util.CartUtils;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CartService {
+    private final CartRepository cartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final ProductRepository productRepository;
+
+    public CartDTO findActualCartByEmail() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository
+                .findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with email [%s] not found", userEmail))
+                );
+        Cart cart = cartRepository
+                .findByUserIdAndActualTrue(user.getId());
+        CartDTO cartDTO;
+        if (cart == null) {
+            Cart newCart = Cart.builder()
+                    .user(user)
+                    .actual(true)
+                    .cartItems(List.of())
+                    .build();
+            cartRepository.save(newCart);
+            cartDTO = CartUtils.convertToCartDto.apply(newCart);
+            return cartDTO;
+        }
+        cartDTO = CartUtils.convertToCartDto.apply(cart);
+        return cartDTO;
+    }
+
+    public List<CartDTO> getAllCarts() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository
+                .findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with email [%s] not found", userEmail))
+                );
+        return cartRepository
+                .findByUserId(user.getId())
+                .stream()
+                .map(CartUtils.convertToCartDto)
+                .toList();
+    }
+
+    public CartDTO addProductToCart(CartItemRequest cartItemRequest) {
```

Comment:
I suggest to refactor this method in 2 methods with different flow, meaning :
rename - "addOrUpdateCartItemProduct"
so you will have 
1. addToCartItemProduct
2. updateCartItemProduct
and depending on validation, processing of request will be corresponding

Reply:
next time